### DISCUSSION TITLE
Advertisement packet parser

### DISCUSSION THREAD

**Discussion by stavrt** [emericg/toolBLEx#12] [Ideas]
To have the ability to parse the advertisement data, most likely a JS function builder that the payload goes through and can do a simple print log statement. For example, if the payload has 2 bytes Temperature and 1 byte humidity, ect, To be able to parse and print the data and even better, decrypt the data if encrypted, would be great

**Comment by emericg** [comment]
I was thinking about adding support for "generic" parsing of Xiaomi, Qingping and BtHome (v1/v2) advertising formats (with the possibility to add more without too much code for each new format), because I already coded these for my WatchFlower software:
https://github.com/emericg/WatchFlower/blob/master/src/device_sensor_advertisement.cpp

But this is not a particularly easy feature to do right. And because virtually every device has its own format, the usefulness would be low (but it would still be pretty cool).

**Comment by deepcoder** [comment]
You might have a look at the work these folks are doing to decode advertising packets, it would be interesting to have an open source 'lingua franca' library of decoders across multiple projects.

https://decoder.theengs.io/participate/adding-decoders.html
